src/nomad_simulations/schema_packages/workflow/general.py

```diff
@@ -19,6 +20,65 @@ class SimulationTask(Task):
     pass
 
 
+class WorkflowConvergenceTarget(ArchiveSection):
+    """
+    A section for defining convergence targets.
+
+    Calculations are converged w.r.t. to different parameters. This section holds the parameter name
+    and the unit of the convergence parameter.
+    """
+
+    # TODO: This most probably needs more explanation, and/or needs to be changed to an Enum to reduce ambiguity when populating.
+    convergence_parameter_name = Quantity(
+        type=str,
+        description="""
+        Name of the parameter that is converged, e.g., energy, forces, electron density, ...
+        """
+    )
+
+    convergence_threshold = Quantity(
+        type=np.float64,
+        description="""
+        Numerical value of the parameter that is converged.
+        """
+    )
+
+    # This is copied from PR #191 https://github.com/nomad-coe/nomad-simulations/pull/191
+    threshold_type = Quantity(
+        type=MEnum('absolute', 'relative', 'maximum', 'rms', 'residuum'),
+        description="""
+            Specifies the mathematical method used to evaluate convergence between successive self-consistent field (SCF) iterations. 
+    This determines how differences between iterations are calculated and compared against the convergence threshold.
+    
+    The available comparison modes are:
+    
+    | Mode | Description |
+    | --------- | -------------------------------- |
+    | `'absolute'` | Measures the absolute difference between two subsequent iterations (e.g., |E_n - E_{n-1}|). Most common for energy convergence. |
+    | `'relative'` | Calculates the relative difference as a fraction of the total property value (e.g., |E_n - E_{n-1}|/|E_n|). Useful when the magnitude of the property varies widely across systems. |
+    | `'residuum'` | Computes the absolute difference between the current value and the value estimated from the wavefunction at the start of the step. Often used for evaluating convergence of the electron density. |
+    | `'maximum'` | Reports the maximum absolute difference across all components of a multi-component property (e.g., max|F_i,n - F_i,{n-1}| for forces). Suitable for vector quantities like forces or stress tensor elements. |
+    | `'rms'` | Calculates the root mean square of differences across all components (e.g., √(∑|F_i,n - F_i,{n-1}|²/N)). Provides a statistical measure of overall convergence for multi-component properties. |
+    
+    The mode used affects both convergence behavior and computational efficiency. Different codes may default to different comparison modes for the same physical property.
+        """,
+    )
+
+    convergence_threshold_unit = Quantity(
+        type=str,
+        description="""
+        Unit using the pint UnitRegistry() notation for the `convergence_threshold`.
+        """
+    )
```

Comment:
Note that this won't be compatible with the NOMAD framework for handling units. It won't convert when the user asks so. Moreover, to fit your description, you should alter the `type`. This still won't power the unit conversion on request.

The modular handling of units is very hard. Many ppl have given it a shot. The closest we have right now are these dataframes.